src/crc_prediction_markets/BetContractFactory.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./BetContract.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+// Struct to store market information
+struct MarketInfo {
+    address fpmmAddress;
+    address groupCRCToken;
+    uint256[] outcomeIdxs;
+    address[] betContracts;
+}
+
+contract BetContractFactory {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    address public hubAddress;
+
+    constructor(address _hubAddress) {
+        hubAddress = _hubAddress;
+    }
+
+    event BetContractDeployed(address indexed betContract, uint256 indexed outcomeIndex);
+
+    // Mapping from FPMM address to market information
+    mapping(address => MarketInfo) public fpmmToBetContracts;
+
+    EnumerableSet.AddressSet private fpmmAddresses;
+
+    // Public function to return all processed FPMM addresses
+    function getAllProcessedFPMMAddresses() external view returns (address[] memory) {
+        return fpmmAddresses.values();
+    }
+
+    function getMarketInfo(address fpmmAddress) public view returns (MarketInfo memory) {
+        return fpmmToBetContracts[fpmmAddress];
+    }
+
+    function fpmmAlreadyProcessed(address fpmmAddress) public view returns (bool) {
+        return fpmmAddresses.contains(fpmmAddress);
+    }
+
+    function createBetContract(
+        address fpmmAddress,
+        address groupCRCToken,
+        uint256 outcomeIndex,
+        address hubAddress,
+        uint256 betContractIdentifier
+    ) private returns (address) {
+        BetContract betContract =
+            new BetContract(fpmmAddress, groupCRCToken, outcomeIndex, hubAddress, betContractIdentifier);
+        emit BetContractDeployed(address(betContract), outcomeIndex);
+
+        return address(betContract);
+    }
+
+    function createBetContractsForFpmm(address fpmmAddress, address groupCRCToken, uint256[] memory outcomeIndexes)
+        external
+    {
+        // Create market if it doesn't exist
+        if (!fpmmAddresses.contains(fpmmAddress)) {
+            // We assume binary markets, hence 2 outcomes
```

Comment:
That's a shame after the big categorical PR coming in, would it be hard to iterate all outcomes of market here automatically? (it kind of looks like it's being done already actually)

Reply:
The comment is wrong, deleting and closing this. Agree 100%